controllers/storagecluster/external_resources.go

```diff
@@ -326,6 +339,105 @@ func (obj *ocsExternalResources) ensureDeleted(r *StorageClusterReconciler, inst
 	return reconcile.Result{}, nil
 }
 
+// claims should be created only once and should not be created/updated again if user deletes/update it.
+func (r *StorageClusterReconciler) createDefaultStorageClassClaims(instance *ocsv1.StorageCluster) error {
+
+	StorageClassClaimFile := &ocsv1alpha1.StorageClassClaim{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      generateNameForCephFilesystemSC(instance),
+			Namespace: instance.Namespace,
+			Labels: map[string]string{
+				defaultStorageClassClaimLabel: "true",
+			},
+		},
+		Spec: ocsv1alpha1.StorageClassClaimSpec{
+			Type: "sharedfilesystem",
+		},
+	}
+
+	StorageClassClaimBlock := &ocsv1alpha1.StorageClassClaim{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      generateNameForCephBlockPoolSC(instance),
+			Namespace: instance.Namespace,
+			Labels: map[string]string{
+				defaultStorageClassClaimLabel: "true",
+			},
+		},
+		Spec: ocsv1alpha1.StorageClassClaimSpec{
+			Type: "blockpool",
+		},
+	}
+
+	err := r.createAndOwnStorageClassClaim(instance, StorageClassClaimFile)
+	if err != nil {
+		return err
+	}
+
+	err = r.createAndOwnStorageClassClaim(instance, StorageClassClaimBlock)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (r *StorageClusterReconciler) createAndOwnStorageClassClaim(
+	instance *ocsv1.StorageCluster, claim *ocsv1alpha1.StorageClassClaim) error {
+
+	err := controllerutil.SetOwnerReference(instance, claim, r.Client.Scheme())
+	if err != nil {
+		return err
+	}
+
+	err = r.Client.Create(context.TODO(), claim)
+	if err != nil && !errors.IsAlreadyExists(err) {
+		return err
+	}
+
+	return nil
+}
+
+func (r *StorageClusterReconciler) deleteDefaultStorageClassClaims(instance *ocsv1.StorageCluster) error {
+
+	storageClassClaims := &ocsv1alpha1.StorageClassClaimList{}
+	err := r.Client.List(context.TODO(), storageClassClaims, &client.ListOptions{
+		Raw: &metav1.ListOptions{LabelSelector: defaultStorageClassClaimLabel}})
+	if err != nil {
+		return err
+	}
+
+	for _, storageClassClaim := range storageClassClaims.Items {
+
+		err = r.Client.Delete(context.TODO(), &storageClassClaim)
+		if err != nil && !errors.IsNotFound(err) {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (r *StorageClusterReconciler) verifyNoStorageClassClaimsExist(instance *ocsv1.StorageCluster) error {
+
+	storageClassClaims := &ocsv1alpha1.StorageClassClaimList{}
+	err := r.Client.List(context.TODO(), storageClassClaims)
+	if err != nil {
+		return err
+	}
+
+	for _, storageClassClaim := range storageClassClaims.Items {
+		if _, ok := storageClassClaim.Labels[defaultStorageClassClaimLabel]; !ok {
+			err = inbuilterrors.New("Failed to cleanup resources. storageClassClaims are present." +
+				"Delete all storageClassClaims for the cleanup to proceed")
```

Comment:
Why not use `fmt.Errorf` which is more wildly known?

Reply:
because of this comment https://github.com/red-hat-storage/ocs-operator/pull/1607#discussion_r840252417